src/main/java/com/github/mc1arke/sonarqube/plugin/ce/pullrequest/AnalysisDetails.java

```diff
@@ -116,20 +116,20 @@ public String getCommitSha() {
 
     public String createAnalysisSummary(FormatterFactory formatterFactory) {
 
-        BigDecimal newCoverage =
-                findQualityGateCondition(CoreMetrics.NEW_COVERAGE_KEY)
-                    .filter(condition -> condition.getStatus() != EvaluationStatus.NO_VALUE)
-                    .map(QualityGate.Condition::getValue)
-                    .map(BigDecimal::new)
-                    .orElse(null);
+        BigDecimal newCoverage = findMeasure(CoreMetrics.NEW_COVERAGE_KEY)
+                .map(Measure::getDoubleValue)
+                .map(BigDecimal::new)
+                .orElse(null);
```

Comment:
Ok, this doesn't work; it throws `java.lang.IllegalStateException: null`. But the previous way to get to get the new coverage value (via `findQualityGateCondition`) always returns 0% even in cases where it isn't.

@mc1arke Do you have any idea what the correct way to get the new coverage percentage?

Reply:
To clarify: getting the new coverage via `findQualityGateCondition` will yield 0% if coverage is not a condition on new code. If it is, the it will return the correct value.